### ISSUE TITLE
Check Enum.create_editor; add tests

### ISSUE THREAD

**Issue by mdickinson** [enthought/traits#965]
I'm suspicious of the code in `Enum.create_editor`, and it isn't exercised by the test suite. We should double check that it works as intended before the 6.1.0 release (and ideally, also add tests).

We should also check that the editor behaves sensibly for the various possible input types (which changed in #889).

For example: is that `values = self` *really* right? For another, there's some code that looks editor-related in the `BaseEnum` `__init__` method, but that code never gets executed, and doesn't belong in the `__init__` in the first place.

**Comment by midhun-pm** [comment]
The current code on master does work for `Enum`'s with collection arguments.
The only case I could find that caused problems was `Enum([1, {1, 2}])` which errored ou. 
This is a valid `Enum` declaration where the valid values are either `1` or the set `{1,2}` . Is this something that we want to fix?

**Comment by mdickinson** [comment]
That depends. What was the error message, and what was its underlying cause?

**Comment by midhun-pm** [comment]
The exact cause of the bug is trying to use the `set` `{1,2}`  as the key in a `dict`  in [traitsui.helper](https://github.com/enthought/traitsui/blob/80fa6fb6c9158cb3d27ecf6243751f313a02902b/traitsui/helper.py#L101) when trying to create an inverse mapping from the value to its string representation. I believe this is to update the items in the editor in case the value changes externally. 

I verified that the editor works fine if the exception is caught, but this will break the automatic update on external value changes. Shall I open an issue in `traitsui` ?

**Comment by midhun-pm** [comment]
However, I can't think of good common use cases where we would want an enumeration of different kind of types.

**Comment by kitchoi** [comment]
> is that values = self really right?

This is the code that allows `Enum` to be passed directly to `EnumFactory`:
https://github.com/enthought/traitsui/blob/b7a409d4e869d1857fcf77e8be47de581e864d41/traitsui/helper.py#L91-L92

This is after a bunch of ifs fall through. I think we should just set `values` to `self.values` in `Enum.create_editor`.

Off-topic: When I was reviewing #988 and looking into this issue, I discovered this traitsui issue: https://github.com/enthought/traitsui/issues/782

**Comment by kitchoi** [comment]
> When I was reviewing #988 and looking into this issue, I discovered this traitsui issue: enthought/traitsui#782

https://github.com/enthought/traitsui/issues/782 is not entirely orthogonal: because if we want the `Enum.create_editor` to create an editor that behaves correctly, then we would have to workaround https://github.com/enthought/traitsui/issues/782 by assigning the value after the editor is instantiated:
```
        editor = EnumEditor(
            values=values,
            name=name,
            cols=self.cols or 3,
            evaluate=self.evaluate,
            format_func=self.format_func,
            mode=self.mode if self.mode else "radio",
        )
        if self.name is None:
            # workaround enthought/traitsui#782
            editor.values = values
        return editor
```

I'd think this `format_func` and `evaluate` should not be passed via `traits.Enum`... but that's a separate discussion.

**Comment by mdickinson** [comment]
Closed by #988.
